libdnf/hy-iutil.c

```diff
@@ -531,6 +531,60 @@ what_downgrades(Pool *pool, Id pkg)
     return l;
 }
 
+char *
+pool_solvable_epoch_optional_2str(Pool *pool, Solvable *s, gboolean with_epoch)
+{
+    const char *e;
+    const char *name = pool_id2str(pool, s->name);
+    const char *evr = pool_id2str(pool, s->evr);
+    const char *arch = pool_id2str(pool, s->arch);
+    gboolean present_epoch = FALSE;
+
+    for (e = evr + 1; *e != '-' && *e != '\0'; ++e) {
+        if (*e == ':') {
+            present_epoch = TRUE;
+            break;
+        }
+    }
+    char *output_string;
+    int name_length, evr_length, arch_length;
+    int extra_epoch_length = 0;
+    name_length = strlen(name);
+    evr_length = strlen(evr);
+    arch_length = strlen(arch);
+    if (!present_epoch && with_epoch) {
+        extra_epoch_length = 2;
+    } else if (present_epoch && !with_epoch) {
+        extra_epoch_length = evr - e - 1;
+    }
+
+    output_string = pool_alloctmpspace(
+        pool, name_length + evr_length + extra_epoch_length + arch_length + 3);
+
+    strcpy(output_string, name);
+
+    if (evr_length || extra_epoch_length)
```

Comment:
May be
`if (evr_length || extra_epoch_length > 0)`

Reply:
extra_epoch_length > 0 IS MORE READABLE BUT the original expretion has same meaning according to previous code

Reply:
Changed